Source/EasyNetQ/Consumer/DefaultConsumerErrorStrategy.cs

```diff
@@ -92,17 +93,18 @@ private void Connect()
             }
         }
 
-        private void DeclareDefaultErrorQueue(IModel model)
+        private void DeclareDefaultErrorQueue(IModel model, ConsumerExecutionContext context)
         {
-            if (!errorQueueDeclared)
+            var errorQueueName = conventions.ErrorQueueNamingConvention(context.Info);
+            if (!errorQueuesDeclared.Contains(errorQueueName))
```

Comment:
don't have time to review this deeply, but could this be open a concurrent issue? is this piece of code reachable from more than one thread at time? If yes, need to be thread-safe.